src/zenml/integrations/github/secrets_managers/github_secrets_manager.py

```diff
@@ -0,0 +1,363 @@
+
+    @property
+    def post_registration_message(self) -> Optional[str]:
+        """Info message regarding the required environment variables to
+        authenticate with the GitHub API."""
+        return AUTHENTICATION_CREDENTIALS_MESSAGE
+
+    @property
+    def session(self) -> requests.Session:
+        """Session to send requests to the GitHub API."""
+        if not self._session:
+            session = requests.Session()
+            github_username = os.getenv(ENV_GITHUB_USERNAME)
+            authentication_token = os.getenv(ENV_GITHUB_AUTHENTICATION_TOKEN)
+
+            if not github_username or not authentication_token:
+                raise RuntimeError(
+                    "Missing authentication credentials for GitHub secrets "
+                    "manager. " + AUTHENTICATION_CREDENTIALS_MESSAGE
+                )
+
+            session.auth = HTTPBasicAuth(github_username, authentication_token)
+            session.headers["Accept"] = "application/vnd.github.v3+json"
+            self._session = session
+
+        return self._session
+
+    def _send_request(
+        self, method: str, resource: Optional[str] = None, **kwargs: Any
+    ) -> requests.Response:
+        """Sends an HTTP request to the GitHub API.
+
+        Args:
+            method: Method of the HTTP request that should be sent.
+            resource: Optional resource to which the request should be sent. If
+                none is given, the default GitHub API secrets endpoint will be
+                used.
+            **kwargs: Will be passed to the `requests` library.
+
+        Returns:
+            HTTP response.
+
+        Raises:
+            HTTPError: If the request failed due to a client or server error.
+        """
+        url = (
+            f"https://api.github.com/repos/{self.owner}/{self.repository}"
+            f"/actions/secrets"
+        )
+        if resource:
+            url += resource
+
+        response = self.session.request(method=method, url=url, **kwargs)
+        # Raise an exception in case of a client or server error
+        response.raise_for_status()
+
+        return response
+
+    def _encrypt_secret(self, secret_value: str) -> Tuple[str, str]:
+        """Encrypts a secret value.
+
+        This method first fetches a public key from the GitHub API and then uses
+        this key to encrypt the secret value. This is needed in order to
+        register GitHub secrets using the API.
+
+        Args:
+            secret_value: Secret value to encrypt.
+
+        Returns:
+            The encrypted secret value and the key id of the GitHub public key.
+        """
+        import base64
+
+        from nacl.encoding import Base64Encoder
+        from nacl.public import PublicKey, SealedBox
+
+        response_json = self._send_request("GET", resource="/public-key").json()
+        public_key = PublicKey(
+            response_json["key"].encode("utf-8"), Base64Encoder
```

Comment:
what if response.json is empty? or key is not there?

Reply:
The `_send_request` method makes sure the response code is `20*`, in which case the schema and everything is guaranteed to look like this by the github api.